pkg/codegen/dotnet/gen_program.go

```diff
@@ -280,7 +280,12 @@ func GenerateProject(directory string, project workspace.Project, program *pcl.P
 		if langInfo, found := p.Language["csharp"]; found {
 			csharpInfo, ok := langInfo.(CSharpPackageInfo)
 			if ok {
-				namespace := namespaceName(csharpInfo.Namespaces, p.Name)
+				pkgName := p.Name
+				if name := csharpInfo.PackageName; name != "" {
```

Comment:
You'll need to add this field to the struct here:

https://github.com/pulumi/pulumi/blob/217088236a2ab923884e940fa18da9750f6e2fbc/pkg/codegen/dotnet/importer.go#L29